amethyst_animation/src/systems/control.rs

```diff
@@ -447,14 +447,15 @@ where
     T: AnimationSampling + Component + Clone,
 {
     // check that hierarchy is valid, and all samplers exist
-    if animation
+    let valid = animation
         .nodes
         .iter()
-        .any(|&(ref node_index, _, ref sampler_handle)| {
```

Comment:
Wouldn't any be preferred here? All will make it go through all of them even if it's not necessary